File: app/src/main/java/com/example/cinemates/adapter/PersonViewHolder.java
package com.example.cinemates.adapter;

import androidx.annotation.NonNull;

import com.example.cinemates.databinding.ListItemPersonLongBinding;
import com.example.cinemates.databinding.ListItemPersonSmallBinding;
import com.example.cinemates.util.RecyclerViewEmptySupport;

/**
 * @author Antonio Di Nuzzo
 * Created 28/07/2022 at 08:51
 */
class PersonViewHolder extends RecyclerViewEmptySupport.ViewHolder {
    ListItemPersonLongBinding mLongBinding;
    ListItemPersonSmallBinding mSmallBinding;

    PersonViewHolder(@NonNull ListItemPersonLongBinding listItemPersonLongBinding) {
        super(listItemPersonLongBinding.getRoot());
        this.mLongBinding = listItemPersonLongBinding;

    }

    PersonViewHolder(@NonNull ListItemPersonSmallBinding listItemPersonSmallBinding) {
        super(listItemPersonSmallBinding.getRoot());
        this.mSmallBinding = listItemPersonSmallBinding;


    }


}

File: app/src/main/java/com/example/cinemates/adapter/PersonViewHolder.kt
package com.example.cinemates.adapter

import androidx.recyclerview.widget.RecyclerView
import com.example.cinemates.databinding.ListItemPersonLongBinding
import com.example.cinemates.databinding.ListItemPersonSmallBinding

/**
 * @author Antonio Di Nuzzo
 * Created 28/07/2022 at 08:51
 */
internal class PersonViewHolder : RecyclerView.ViewHolder {
    var longBinding: ListItemPersonLongBinding? = null
    var smallBinding: ListItemPersonSmallBinding? = null

    constructor(listItemPersonLongBinding: ListItemPersonLongBinding) : super(
        listItemPersonLongBinding.root
    ) {
        longBinding = listItemPersonLongBinding
    }

    constructor(listItemPersonSmallBinding: ListItemPersonSmallBinding) : super(
        listItemPersonSmallBinding.root
    ) {
        smallBinding = listItemPersonSmallBinding
    }
}